definitions/marts/looker_studio_marts/ecf/ls_ecf_sandbox_api_monitoring.sqlx

```diff
@@ -0,0 +1,96 @@
+    },
+    bigquery: {
+        partitionBy: "DATE(occurred_at)",
+        clusterBy: ["request_method", "user_id", "response_status"]
+    },
+    description: "This mart produces a table outlining all Testing API calls that have been made by Lead Providers in the sandbox testing environment. It excludes internal users (identified as those with a NULL user_id) to ensure the focus is more on LP usage. The table outlines the response statuses of those calls to help the LPDOB and Digital Engagement team identify how the LPs are engaging with the new testing API. The final output feeds into the Testing API tabs of the CPD- API Monitoring dashboard.",
+    columns: {
+        occurred_at: "Timestamp of when the API call was made.",
+        user_id: "User ID of the Lead Provider making the API call. NULL user_ids are excluded as they are internal users. These user_ids differ from the user_ids available for LPs in the Production environment and cannot be mapped onto the cpd_lead_provider_id as user_ids for within the Production environment can",
+        request_uuid: "ID of individual request.",
+        request_method: "Method for calling the API. Can be GET, POST, or PUT.",
+        request_path: "The endpoint that is being accessed.",
+        request_query: "Populated by any filters used when making a GET call.",
+        response_status: "The code associated with the API call. Any non-200 response status is an error.",
+        lp_name: "Name of the Lead Provider making the API call.",
+        request_body: "Details the call made by the Lead Provider.",
+        response_body: "Details the API response. Usually populated if there are any errors with the API call being made.",
+        hidden_DATA: {
+            columns: {
+                key: "Name of a field containing sensitive data included in this API request custom event from dfe-analytics",
+                value: {
+                    description: "Contents of this field",
+                    bigqueryPolicyTags: ["projects/ecf-bq/locations/europe-west2/taxonomies/6302091323314055162/policyTags/301313311867345339"]
+                }
+            }
+        }
+    }
+}
+
+WITH
+  data_cut AS (
+  SELECT
+    * EXCEPT (anonymised_user_agent_and_ip,
+      DATA,
+      entity_table_name,
+      environment,
+      event_tags,
+      event_type,
+      namespace,
+      request_referer,
+      request_user_agent,
+      response_content_type)
+  FROM
+    ${ref("ecf_events_sandbox", "events")}
+  WHERE
+    event_type = 'web_request'
+    AND user_id IS NOT NULL
+    AND request_path LIKE '/api/v%'
+    AND request_uuid NOT IN ('edc376f322a4dc273b62997804695fb5',
+      '13f9f767fabfcb2cafc16f366b72660e',
+      '3b0872b43cd564d2133dbbff6fa1a4cf',
+      'ad6a40fda446b96f3212cdb57d2e98e6',
+      'c1fdb72bd593d9eda54f8e95e9724884',
+      '8bee2f527391c8891cbaee1775670f46') ),
+  request_response_fields AS (
+  SELECT
+    eep.request_uuid,
+    STRING_AGG(DISTINCT

```

Comment:
You can use ```${data_functions.eventDataExtract(...)}``` for this I believe. See https://github.com/DFE-Digital/dfe-analytics-dataform/blob/master/includes/data_functions.js for details.